docs/CodingStandards.md

```diff
@@ -0,0 +1,165 @@
+obvious that you are running the default constructor, perhaps by appending `{}` to the
+variable name. This is considered unnecessary for well-known types with a sensible default
+constructor, such as the standard library and LLVM container types.
+
+## Braces
+
+Brace placement is covered by clang-format below.
+
+C++ allows the body of an if, else, while, do, for to be a single simple statement,
+without any braces. We follow the LLVM convention of only using an unbraced body if
+it is a single line, including comments. Otherwise the body should be braced. For
+an "if..else", if one of the two bodies is braced, the other one should be too. For
+an "if..else if..else if..else", if any body is braced, the others should be too.
+
+## Redundant comparisons
+
+Comparison of a bool with `false` or a pointer with `nullptr` is redundant and should
+be avoided. Use `!` instead where appropriate.
+
+## Redundant parentheses
+
+Certain cases of theoretically redundant parentheses around a subexpression are needed
+because compilers warn about them otherwise, for the good reason that C/C++ operator
+precedence makes it difficult to understand and easy to get wrong (in particular
+`&` `|` `^` vs comparison operators, and `&&` vs `||`). However redundant parentheses
+are not needed in cases that compilers do not warn about, in particular a comparison
+inside `&&` or `||`, or a comparison used as the condition of `?:` or in an assignment
+or `return`.
+
+## Comments
+
+Despite the look of much LLVM code, the LLVM coding standard does actually recommend
+placing a comment at the top of a function/method. :-)
+
+In LLPC, that is mandatory. Parameters should be commented in the main comment at
+the top of the function like this:
+
+```
+// \param hatstand : [in/out] The Hatstand object to modify
+```
+
+A tag of `[in]` is unnecessary, and is assumed for a reference or pointer parameter
+where no tag is given.
+
+Parameter comments should not be placed on the parameter line itself and aligned
+(as happens in PAL style), to avoid spuriously large diffs when a change or addition
+of one parameter means all the others need realigning.
+
+Similarly, a bunch of fields or variables should not have aligned inline comments.
+Instead, put each field's comment on the line above the field.
+
+## clang-format
+
+The LLPC coding standard uses a clang-format configuration based on the LLVM one,
+with the following changes:
+
+- 120 column limit (rather than 80)
+- `#include` blocks are merged together (removing blank lines) before sorting
+- `#include` directives are sorted in this order:
+  - the `.h` file for the current `.cpp` file;
+  - all other ones;
+  - `lgc/*` files;
+  - `llvm/*` files;
+  - standard library.
+
+You should use clang-format to format code modified or added in your change, but do not
+reformat any other code. Reviewers should reject a change that has spurious reformatting
```

Comment:
Note: a CI bot could be added that checks for that in the future, to reduce the turnaround times.